### ISSUE TITLE
Replace Electron with Tauri

### ISSUE THREAD

**Issue by benonymity** [divamgupta/diffusionbee-stable-diffusion-ui#354]
To reduce the large size of the app could it switch to [Tauri](https://tauri.app)? It does the same thing as Electron, but is written in Rust and far more lightweight.

**Comment by arunavo4** [comment]
Looking at this [comparison](https://www.levminer.com/blog/tauri-vs-electron) It will be very hard, also there is no python backend support so not sure how the switch will be possible.

**Comment by jingslunt** [comment]
> Looking at this [comparison](https://www.levminer.com/blog/tauri-vs-electron) It will be very hard, also there is no python backend support so not sure how the switch will be possible.

use  pake(tauri tool) too easy. https://github.com/tw93/Pake

**Comment by arunavo4** [comment]
> > Looking at this [comparison](https://www.levminer.com/blog/tauri-vs-electron) It will be very hard, also there is no python backend support so not sure how the switch will be possible.
> 
> use pake(tauri tool) too easy. https://github.com/tw93/Pake

I don't see how will this tool package the python backend, would be very helpful for a simple website / web app which calls external api's

**Comment by arunavo4** [comment]
> Yeah, though Tauri doesn't support different backends, it does support calling files from the app code via the Command API and sidecars, documented [here](https://tauri.app/v1/guides/building/sidecar/) and [here](https://tauri.app/v1/api/js/shell/#command) (though MacOS does come preinstalled with Python, so bundling it would probably be redundant). Probably requires a good deal of work, but the current 150+ MB for essentially just a webpage that downloads and runs StableDiffusion models seems excessive.

The documentation for python does not exist. Discussion [here](https://github.com/tauri-apps/tauri/discussions/2759) 
One Rare [Tweet](https://mobile.twitter.com/TauriApps/status/1532101112937218048) on it.

**Comment by meszaroszoltan** [comment]
Yes, please rewrite the whole thing _in Rust™_  ASAP because Electron bad 
/s
